kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -108,7 +108,7 @@ String filterDescriptor() {
     @Override
     public void channelActive(ChannelHandlerContext ctx) throws Exception {
         this.ctx = ctx;
-        this.promiseFactory = new PromiseFactory(ctx.executor(), timeoutMs, TimeUnit.MILLISECONDS, LOGGER.getName());
+        this.promiseFactory = new PromiseFactory(ctx.channel().eventLoop(), timeoutMs, TimeUnit.MILLISECONDS, LOGGER.getName());
```

Comment:
what's the difference here?

Reply:
for some reason I thought that only EventLoop had the `inEventLoop()` type methods, looks like the EventExecutor would be fine here, will revert

Reply:
have reverted this as part of the refactor to use `ThreadAwareExecutor` instead